lib/grakn/cache.ex

```diff
@@ -4,51 +4,83 @@ defmodule Grakn.Cache do
   for performance, so we share sessions for the same keyspace.
   """
 
-  import Cachex.Spec
-  @cache __MODULE__
+  use GenServer
+
+  import Record
```

Comment:
One question and one observation: 

question: did you go for Record just because it is, in this case, faster in reads? 
observation: why not using `import Record, only: [defrecord: 2]` here instead?

Reply:
I go for records not because of access performance, but because of interaction with ets. 

It is a default(and only supported, tuple or record) structure, which you can query with match spec in ets. For example, see how `expired` implemented:

```
  def expired() do
    select_spec =
      fun do
        entry(ttl: ttl, touched: touched) = entry_elem when touched + ttl < ^now() -> entry_elem
      end

    :ets.select(@table, select_spec)
  end
```

It use select, which will only return outdated elements from ets table. If the structure was saved in map, than you need to go over every element manually(with fold) to determinate, which are elements outdated. Selecting elements from ets is quicker and less resource intensive, as copy whole ets to a process and doing it manually (in case of select, only found elements will be copied to a process).

Probably, in every case, where you need to construct any match specification against `ets`, records are still the way to go.